### PR TITLE
Hadnle admin functionalities

### PR DESCRIPTION
**Pull request by motaz99**

### CODE REVIEW

**Comment by motaz99** [comment]
https://vimeo.com/1030298945/4d6a0eaf20?share=copy

File: src/pages/api/signup.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    const { username, email, password, role = "admin" } = req.body;
+
+    if (!username || !email || !password) {
+      return res
+        .status(400)
+        .json({ error: "Username, email, and password are required." });
+    }
+
+    try {
+      const hashedPassword = await bcrypt.hash(password, 10);
```
Comment (MohammedAlabd) [inline src/pages/api/signup.js]: take a look at this

https://www.quora.com/If-I-hash-passwords-before-storing-them-in-my-database-is-that-sufficient-to-prevent-them-being-recovered-by-anyone

also search about this " is hashing the password before storing it in the DB sufficient?"

Comment (MohammedAlabd) [reply src/pages/api/signup.js]: check this as well
https://markilott.medium.com/password-storage-basics-2aa9e1586f98

File: src/pages/api/auth/login.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+      if (!user) {
+        return res.status(401).json({ error: "Invalid email or password" });
+      }
+      const isPasswordValid = await bcrypt.compare(password, user.password);
+      if (!isPasswordValid) {
+        return res.status(401).json({ error: "Invalid email or password" });
+      }
+
+      const token = generateToken(user);
```
Comment (MohammedAlabd) [inline src/pages/api/auth/login.js:29]: don't you need to save the token somewhere?

**Review by MohammedAlabd — CHANGES_REQUESTED** [review]

File: src/pages/api/auth/login.js (hunk 2 above)
Comment (motaz99) [reply src/pages/api/auth/login.js:29]: I am storing it in the Cookie, isn't that enough?  This is the way I usually do it, storing the token in the cookie and then later with each request from the user that has logged in, I will send the token and they decode it so I know which user is this. This is what I know, but is there other things that I don't know? I would like to learn.

@MohammedAlabd

**Pull request merged**